src/choreo/vote/fd_vote.c

```diff
@@ -0,0 +1,117 @@
+#include "fd_vote.h"
+
+#pragma GCC diagnostic ignored "-Wformat"
+#pragma GCC diagnostic ignored "-Wformat-extra-args"
+
+ulong
+fd_vote_txn_generate(fd_compact_vote_state_update_t *compact_vote_update,
+                     fd_pubkey_t *validator_pubkey,
+                     fd_pubkey_t *vote_acct_pubkey,
+                     uchar* validator_privkey,
+                     uchar* vote_acct_privkey,
+                     uchar* recent_blockhash,
+                     uchar out_txn_meta_buf [static FD_TXN_MAX_SZ],
+                     uchar out_txn_buf [static FD_TXN_MTU]
+                     ) {
+  /* Create the transaction base */
+  fd_pubkey_t pubkeys[2], vote_program_pubkey;
+  memcpy( pubkeys, validator_pubkey, sizeof(fd_pubkey_t) );
+  memcpy( pubkeys + 1, vote_acct_pubkey, sizeof(fd_pubkey_t) );
+  memcpy( &vote_program_pubkey, &fd_solana_vote_program_id, sizeof(fd_pubkey_t) );
+
+  fd_txn_accounts_t vote_txn_accounts;
+  vote_txn_accounts.signature_cnt         = 2;
+  vote_txn_accounts.readonly_signed_cnt   = 0;
+  vote_txn_accounts.readonly_unsigned_cnt = 1;
+  vote_txn_accounts.acct_cnt              = 3;                    /* 3 accounts: validator, vote account, vote program  */
+  vote_txn_accounts.signers_w             = pubkeys;              /* 2 signers: validator, vote account */
+  vote_txn_accounts.signers_r             = NULL;
+  vote_txn_accounts.non_signers_w         = NULL;
+  vote_txn_accounts.non_signers_r         = &vote_program_pubkey; /* 1 non-signer: vote program */
+  FD_TEST( fd_txn_base_generate( out_txn_meta_buf, out_txn_buf, 2, &vote_txn_accounts, recent_blockhash ) );
+
+  /* Add the vote instruction to the transaction */
+  fd_vote_instruction_t vote_instr;
+  uchar vote_instr_buf[FD_TXN_MTU];
+  vote_instr.discriminant = fd_vote_instruction_enum_compact_update_vote_state;
+  vote_instr.inner.compact_update_vote_state = *compact_vote_update;
+  fd_bincode_encode_ctx_t encode = { .data = vote_instr_buf, .dataend = (vote_instr_buf + FD_TXN_MTU) };
+  fd_vote_instruction_encode ( &vote_instr, &encode );
+  ushort vote_instr_size = (ushort)fd_vote_instruction_size( &vote_instr );
+
+  uchar instr_accounts[2];
+  instr_accounts[0] = 1;  /* vote account   */
```

Comment:
both instr accounts are the same?

Reply:
This is the case for local agave validator. Testnet does have 2 different accounts and I will work on adapting to testnet validator next.

Reply:
I know this PR is old, but I wish to let you know that your concern here has been addressed in my latest PR #2059